File: Order/src/main/java/com/badasstechie/order/model/Order.java
package com.badasstechie.order.model;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.PastOrPresent;
import lombok.*;
import java.time.Instant;
import java.util.List;

import static javax.persistence.GenerationType.IDENTITY;

@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "order_table") // order is a reserved keyword in SQL
public class Order {
    @Id
    @GeneratedValue(strategy = IDENTITY)
    private Long id;

    @NotNull(message = "User id is required")
    private Long userId;

    @NotBlank(message = "Order number is required")
    private String orderNumber;

    @OneToMany(cascade = CascadeType.ALL)
    @ToString.Exclude
    private List<OrderItem> items;

    @Enumerated(EnumType.STRING)
    private OrderStatus status;

    @NotBlank(message = "Delivery address is required")
    private String deliveryAddress;

    @PastOrPresent(message = "Created date must be in the past or present")
    private Instant created;
}

File: Order/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Order.Models;

[Table("order_table")]
public class Order
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public long Id { get; set; }

    [Required]
    public long UserId { get; set; }

    [Required]
    public string OrderNumber { get; set; } = string.Empty;

    public List<OrderItem> Items { get; set; } = new();

    [Required]
    public OrderStatus Status { get; set; } = OrderStatus.PENDING;

    [Required]
    public string DeliveryAddress { get; set; } = string.Empty;

    public DateTime Created { get; set; } = DateTime.UtcNow;
}

public enum OrderStatus
{
    PENDING,
    CONFIRMED,
    SHIPPED,
    DELIVERED,
    CANCELLED
}
